pkg/controller/datavolume-controller.go

```diff
@@ -2599,3 +2623,288 @@ func GetRequiredSpace(filesystemOverhead float64, requestedSpace int64) int64 {
+			{
+				MountPath: common.ImporterDataDir,
+				Name:      DataVolName,
+			},
+		},
+	}
+}
+
+// makeSizeDetectionVolumeSpec creates and returns the size-detection pod's Volume spec
+func makeSizeDetectionVolumeSpec(pvcName string) *corev1.Volume {
+	return &corev1.Volume{
+		Name: DataVolName,
+		VolumeSource: corev1.VolumeSource{
+			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: pvcName,
+				ReadOnly:  false,
+			},
+		},
+	}
+}
+
+// inflateSizeWithOverhead inflates the obtained image size with proper overhead calculations
+func inflateSizeWithOverhead(client client.Client,
+	imgSize string,
+	pvcSpec *corev1.PersistentVolumeClaimSpec,
+	sourceSize *resource.Quantity) (resource.Quantity, error) {
+
+	var returnSize resource.Quantity
+	var err error
+
+	if resolveVolumeMode(pvcSpec.VolumeMode) == corev1.PersistentVolumeFilesystem {
+		fsOverhead, err := GetFilesystemOverheadForStorageClass(client, pvcSpec.StorageClassName)
+		if err != nil {
+			return resource.Quantity{}, err
+		}
+		// Parse filesystem overhead (percentage) into a 64-bit float
+		fsOverheadFloat, err := strconv.ParseFloat(string(fsOverhead), 64)
+		if err != nil {
+			return resource.Quantity{}, err
+		}
+		// Parse the raw string containing the image size into a 64-bit int
+		imgSizeInt, err := strconv.ParseInt(imgSize, 10, 64)
+		if err != nil {
+			return resource.Quantity{}, err
+		}
+		// Merge the previous values into a 'resource.Quantity' struct
+		requiredSpace := GetRequiredSpace(fsOverheadFloat, imgSizeInt)
+		returnSize = *resource.NewScaledQuantity(requiredSpace, 0)
+	} else {
+		// Inflation is not needed with 'Block' mode
+		returnSize, err = resource.ParseQuantity(imgSize)
+		if err != nil {
+			return resource.Quantity{}, err
+		}
+	}
+
+	return returnSize, nil
+}
+
+// updateClonePVCAnnotations updates the clone-related annotations of the source PVC
+func (r *DatavolumeReconciler) updateClonePVCAnnotations(sourcePvc *corev1.PersistentVolumeClaim, virtualSize string) error {
+	currCapacity := sourcePvc.Status.Capacity
+	sourcePvc.Annotations[AnnVirtualImageSize] = virtualSize
+	sourcePvc.Annotations[AnnSourceCapacity] = currCapacity.Storage().String()
+	return r.client.Update(context.TODO(), sourcePvc)
+}
+
+// hasPvcSucceeded returns true if a PVC is in 'Succeeded' phase, false if not
+func hasPvcSucceeded(pvc *corev1.PersistentVolumeClaim) bool {
+	if pvc != nil {
+		phase, ok := pvc.Annotations[AnnPodPhase]
+		if ok && phase == string(corev1.PodSucceeded) {
+			return true
+		}
+	}
+	return false
+}
+
+// hasPodSucceeded returns true if a pod is in 'Succeeded' phase, false if not
+func hasPodSucceeded(pod *corev1.Pod) bool {
+	return pod != nil && pod.Status.Phase == corev1.PodSucceeded
+}
+
+// sizeDetectionPodName returns the name of the size-detection pod accoding to the source PVC's UID
+func sizeDetectionPodName(pvc *corev1.PersistentVolumeClaim) string {
+	return "size-detection-" + string(pvc.UID)
```

Comment:
This works and is technically not wrong, but we use `fmt.Sprintf("size-detection-%s", pvc.UID)` in most of these create name type functions.

Reply:
I used [this other function](https://github.com/kubevirt/containerized-data-importer/blob/main/pkg/controller/datavolume-controller.go#L1315) as a reference, but I can change it if you prefer.